spaghetti/clustering/OpticsClustering.py

```diff
@@ -0,0 +1,50 @@
+"""
+Copyright 2023 Spaghetti team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+language governing permissions and limitations under the License.
+"""
+import numpy as np
+from sklearn.cluster import OPTICS
+from spaghetti.clustering.ClusteringModel import ClusteringModel
+
+class OpticsClustering(ClusteringModel):
+    """
+    OPTICS Clustering model. It is a subclass of the ClusteringModel class.
+    It uses the OPTICS model from sklearn for clustering data.
+    Example usages:
+        For fitting the model:
+            oc = OpticsClustering()
+            oc.fit(data)
+        For predicting clusters:
+            oc = OpticsClustering()
+            clusters = oc.predict(data)
+        For fitting the model and predicting clusters:
+            oc = OpticsClustering()
+            clusters = oc.fit_predict(data)
+
+    :param data: Data to be clustered. It should be a numpy array or similar data structure.
+    """
+    def __init__(self):
+        # Initialize the OPTICS model here
+        self._model: OPTICS = OPTICS()
+
+    def _init_model(self):
```

Comment:
You don't use this private method anywhere.

Reply:
see the comment above